bemani/backend/gf/base.py

```diff
@@ -0,0 +1,351 @@
+from bemani.data import UserID, Score, Data
+from bemani.protocol import Node
+
+class GuitarFreaksBase(CoreHandler, CardManagerHandler, Base):
+    """
+    Trying to work on GFDM right now. now it's time to play the guitar rhythmically :)
+    """
+
+    game = GameConstants.GUITARFREAKS
+
+
+    CHART_TYPE_GUITAR_BEGINNER = 0
+    CHART_TYPE_GUITAR_BASIC = 1
+    CHART_TYPE_GUITAR_ADVANCED = 2
+    CHART_TYPE_GUITAR_EXTREME = 3
+    CHART_TYPE_BASS_BEGINNER = 4
+    CHART_TYPE_BASS_BASIC = 5
+    CHART_TYPE_BASS_ADVANCED = 6
+    CHART_TYPE_BASS_EXTREME = 7
+    CHART_TYPE_OPEN_BEGINNER = 8
+    CHART_TYPE_OPEN_BASIC = 9
+    CHART_TYPE_OPEN_ADVANCED = 10
+    CHART_TYPE_OPEN_EXTREME = 11
+
+    def __init__(self, data: Data, config: Dict[str, Any], model: Model) -> None:
+        super().__init__(data, config, model)
+        if model.rev == 'X':
+            self.omnimix = True
+        else:
+            self.omnimix = False
+
+    @property
+    def music_version(self) -> int:
+        if self.omnimix:
+            return DBConstants.OMNIMIX_VERSION_BUMP + self.version
+        return self.version
+
+    def previous_version(self) -> Optional['GuitarFreaksBase']:
+        """
+        Returns the previous version of the game, based on this game. Should
+        be overridden.
+        """
+        return None
+    
+    def get_profile_by_refid(self, refid: Optional[str]) -> Optional[Node]:
+        """
+        Given a RefID, return a formatted profile node. Basically every game
+        needs a profile lookup, even if it handles where that happens in
+        a different request. This is provided for code deduplication.
+        """
+        if refid is None:
+            return None
+
+        # First try to load the actual profile
+        userid = self.data.remote.user.from_refid(self.game, self.version, refid)
+        profile = self.get_profile(userid)
+        if profile is None:
+            return None
+
+        # Now, return it
+        return self.format_profile(userid, profile)
+
+    def new_profile_by_refid(self, refid: Optional[str], name: Optional[str], locid: Optional[int]) -> Node:
+        """
+        Given a RefID and an optional name, create a profile and then return
+        a formatted profile node. Similar rationale to get_profile_by_refid.
+        """
+        if refid is None:
+            return None
+
+        if name is None:
+            name = 'NONAME'
+
+        # First, create and save the default profile
+        userid = self.data.remote.user.from_refid(self.game, self.version, refid)
+        defaultprofile = ValidatedDict({
+            'name': name,
+            'loc': locid,
+        })
+        self.put_profile(userid, defaultprofile)
+
+        # Now, reload and format the profile, looking up the has old version flag
+        profile = self.get_profile(userid)
+        return self.format_profile(userid, profile)
+
+    def put_profile_by_refid(self, refid: Optional[str], request: Node) -> None:
```

Comment:
Same comment here for all _by_refid functions, they appear completely unused. Is this true?